connect/runtime/src/test/java/org/apache/kafka/connect/storage/MemoryConfigBackingStoreTest.java

```diff
@@ -0,0 +1,183 @@
+    private static final List<String> CONNECTOR_IDS = Arrays.asList("connector1", "connector2");
+
+    // Actual values are irrelevant here and can be used as either connector or task configurations
+    private static final List<Map<String, String>> SAMPLE_CONFIGS = Arrays.asList(
+        Collections.singletonMap("config-key-one", "config-value-one"),
+        Collections.singletonMap("config-key-two", "config-value-two"),
+        Collections.singletonMap("config-key-three", "config-value-three")
+    );
+
+    @Mock
+    private ConfigBackingStore.UpdateListener configUpdateListener;
+    private final MemoryConfigBackingStore configStore = new MemoryConfigBackingStore();
+
+    @Before
+    public void setUp() {
+        configStore.setUpdateListener(configUpdateListener);
+    }
+
+    @Test
+    public void testPutConnectorConfig() {
+        configStore.putConnectorConfig(CONNECTOR_IDS.get(0), SAMPLE_CONFIGS.get(0));
+        ClusterConfigState configState = configStore.snapshot();
+
+        assertTrue(configState.contains(CONNECTOR_IDS.get(0)));
+        // Default initial target state of STARTED should be used if no explicit target state is specified
+        assertEquals(TargetState.STARTED, configState.targetState(CONNECTOR_IDS.get(0)));
+        assertEquals(SAMPLE_CONFIGS.get(0), configState.connectorConfig(CONNECTOR_IDS.get(0)));
+        assertEquals(1, configState.connectors().size());
+
+        verify(configUpdateListener).onConnectorConfigUpdate(eq(CONNECTOR_IDS.get(0)));
+    }
+
+    @Test
+    public void testPutConnectorConfigUpdateExisting() {
+        configStore.putConnectorConfig(CONNECTOR_IDS.get(0), SAMPLE_CONFIGS.get(0));
+        ClusterConfigState configState = configStore.snapshot();
+
+        assertTrue(configState.contains(CONNECTOR_IDS.get(0)));
+        // Default initial target state of STARTED should be used if no explicit target state is specified
+        assertEquals(TargetState.STARTED, configState.targetState(CONNECTOR_IDS.get(0)));
+        assertEquals(SAMPLE_CONFIGS.get(0), configState.connectorConfig(CONNECTOR_IDS.get(0)));
+        assertEquals(1, configState.connectors().size());
+
+        configStore.putConnectorConfig(CONNECTOR_IDS.get(0), SAMPLE_CONFIGS.get(1));
+        configState = configStore.snapshot();
+        assertEquals(SAMPLE_CONFIGS.get(1), configState.connectorConfig(CONNECTOR_IDS.get(0)));
+
+        verify(configUpdateListener, times(2)).onConnectorConfigUpdate(eq(CONNECTOR_IDS.get(0)));
+    }
+
+    @Test
+    public void testRemoveConnectorConfig() {
+        configStore.putConnectorConfig(CONNECTOR_IDS.get(0), SAMPLE_CONFIGS.get(0));
+        configStore.putConnectorConfig(CONNECTOR_IDS.get(1), SAMPLE_CONFIGS.get(1));
+        ClusterConfigState configState = configStore.snapshot();
+
+        assertTrue(configState.contains(CONNECTOR_IDS.get(0)));
+        assertTrue(configState.contains(CONNECTOR_IDS.get(1)));
+        assertEquals(2, configState.connectors().size());
```

Comment:
Nit: can do an assertion of the entire expected collection instead of asserting that the collection contains elements x, y, z... and has a certain size.

Reply:
It's about equally verbose for this particular case, but I don't have any strong preference here so I've made this change 👍